lib/improver/weights.py

```diff
@@ -30,123 +30,205 @@
+                coord_exp_vals : string
+                       String list of values which are expected on the
+                       coordinate to be blended over.
+                coord_unit : cf_units.Unit
+                       The unit in which the coord_exp_vals have been passed
+                       in.
+
+            Returns:
+                exp_coord_len: int
+                       The number of forecasts we expect to blend, based on the
+                       length of the coordinate we are going to blend over.
+                exp_forecast_found: binary mask
+                       Array showing where the input cube coordinate values
+                       agree with the input expected coordinate values.
+
+            Raises:
+                ValueError 1 : the coordinate to blend over does not exist on
+                               the cube being blended over.
+                ValueError 2 : the length of the expected coordinate input is
+                               less than the length of the corresponding cube
+                               coordinate.
+                ValueError 3 : the input coordinate units cannot be converted
+                               to the units of the corresponding cube
+                               coordinate.
+        """
+        if not cube.coords(coordinate):
+            msg = ('The coord for this plugin must be '
+                   'an existing coordinate in the input cube.')
+            raise ValueError(msg)
+        cube_coord = cube.coord(coordinate)
+        if coord_exp_vals is not None:
+            coord_values = [float(x) for x in coord_exp_vals.split(',')]
+            if len(coord_values) < len(cube_coord.points):
+                msg = ('The cube coordinate has more points '
+                       'than requested coord, '
+                       'len coord points = {0:d} '.format(len(coord_values)) +
+                       'len cube points = {0:d}'.format(
+                          len(cube_coord.points)))
+                raise ValueError(msg)
+            else:
+                exp_coord = iris.coords.AuxCoord(coord_values,
+                                                 long_name=coordinate,
+                                                 units=coord_unit)
+                exp_coord_len = len(exp_coord.points)
+        else:
+            exp_coord_len = len(cube_coord.points)
+        # Find which coordinates are present in exp_coord but not in cube_coord
+        # ie: find missing forecasts.
+        if len(cube_coord.points) < exp_coord_len:
+            # Firstly check that coord is in the right units
+            # Do not try if coord.units not set
+            if (exp_coord.units != cf_units.Unit('1') and
+                    str(exp_coord.units) != 'no_unit'):
+                if exp_coord.units != cube_coord.units:
+                    try:
+                        exp_coord.convert_units(cube_coord.units)
+                    except:
```

Comment:
except ValueError
So it doesn't capture any other kind of error by mistake. You can still raise the ValueError as you have it.